### PR TITLE
Added environment.yaml and updated dependency versions

### LINKED ISSUE #313: xclim pins xarray version
* xclim version: 0.12.2
* Python version: 3.7
* Operating System: Linux

### Description

xclim pins xarray to 0.13. Currently xarray is at version 0.14.1 - it would be nice if we could use xclim with newer xarray versions.

``` bash

> conda search --info xclim

...
xclim 0.12.2 py_1
-----------------
...
  - xarray 0.13.0
```

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated

### After merging to master and before closing the branch:
- [ ] bumpversion (minor / major / patch) has been called on `master` branch
- [ ] Tags have been pushed

* **What kind of change does this PR introduce?** 
Updates a few pinned dependencies to prevent conda install from having as many issues trying to sort compatibilities of libraries. Also adds a conda `environment.yaml` file for ease of source install.

* **Does this PR introduce a breaking change?** 
No.

* **Other information**:

This PR fixes #313

### CODE REVIEW

**Comment by Zeitsperre** [comment]
@huard @tlogan2000 It looks as though Python3.5 is fully incompatible in xarray 0.14.1. Do we want to push a set of patches to xclim 0.12 and tag a final Python3.5 release before merging this PR?

**Comment by huard** [comment]
I don't think it's necessary.

File: environment.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ - defaults
+dependencies:
+ - scipy>=1.2
+ - numpy>=1.17
+ - pandas>=0.23
+ - cftime==1.0.3.4
+ - netCDF4<1.5.3
+ - dask>=2.8.0
+ - bottleneck>=1.3.1
+ - xarray==0.13.0
```
Comment (huard) [inline environment.yaml]: Unpin xarray ?

**Comment by huard** [comment]
@Zeitsperre Ok for me to unpin xarray and merge ?

**Comment by tlogan2000** [comment]
> @Zeitsperre Ok for me to unpin xarray and merge ?

I think @Zeitsperre is in R. de Loup for Panacées.  

I think the only thing to consdier is whether we want to release a final version supporting py35 that includes all the latest PRs that @aulemahal  and @Zeitsperre have more or less finished  (liquid solide precip accum and poly subsetting being the most interesting).   

We could then unpin and bump minor with this PR?

**Comment by huard** [comment]
I'm :-1: on waiting for these PRs.

**Comment by tlogan2000** [comment]
> I'm  on waiting for these PRs.

Ok for me... I say go for it.   Bye bye py35 hello f-strings

**Comment by Zeitsperre** [comment]
Hey guys, let me look at this tonight. I should be able to push a final version from here. Can we confirm that the History.rst is up to date before I push a final PY35 version?

**Comment by huard** [comment]
The history should mention Pascal's work on rolling. 
It should also more clearly say that `tg` computes tas from tasmin and tasmax.

**Comment by huard** [comment]
@Zeitsperre 
See answer above to your question about changes to history: 
- The history should mention Pascal's work on rolling.
- It should also more clearly say that tg computes tas from tasmin and tasmax.

**Comment by huard** [comment]
Ok. So turns out we're importing xclim in the conf, so mock imports are a no go for now. 
Also, tox does not play nice with conda, so another no go.

**Comment by Zeitsperre** [comment]
@huard The docs build is fixed now in my branch. It turns out that while the GeoPandas stuff that we need for the analyses we're doing require rtree, GeoPandas itself doesn't explicitly need it installed. My approach does an env var check to see whether the "READTHEDOCS" variable exists and add the rtree library to the requirements if it isn't there.

For the anaconda build, I don't feel it's a good approach if we want to have a library that is installable via pip. The dependencies aren't too terrible to configure (aside from rtree and even then, they're working on the macOS build already).

Good to know that that was the reason why autodoc_mock_imports wasn't working. I wasn't sure why that was the case on my side.

It would be good to add a build that uses conda and conda-installed dependencies to see how it interacts with them when pip-installed (ie: when installed with conda binaries with gdal, rtree, proj). Maybe we can add that in here as another test build.

**Comment by huard** [comment]
Please, let's not add anything to that branch. It's been open for a month. Let's merge this ASAP.

**Pull request merged**